.github/workflows/image_build_push.yml

```diff
@@ -0,0 +1,158 @@
+name: docker-image-push-public-dash
+
+on:
+  push:
+    branches: [ main ]
+    
+  workflow_dispatch:
+    inputs:
+      docker_image_tag:
+        description: "Latest Docker image tags passed from e-mission-server repository on image build and push"
+        required: true
+
+env:
+  DOCKER_USER: ${{secrets.DOCKER_USER}}
+  DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+
+jobs:
+  fetch_run_id:
+    runs-on: ubuntu-latest
+
+    outputs:
+      run_id: ${{ steps.get_run_id.outputs.run_id }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with: 
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+         pip install requests
+
+      - name: Run Python script
+        id: run_script
+        run: |
+          echo "Fetching latest successful run ID from e-misison-server docker image workflow"
+          python .github/fetch_runID.py
+    
+      - name: Get Run ID
+        id: get_run_id
+        run: echo "run_id=${{ steps.run_script.outputs.run_id }}" >> "$GITHUB_OUTPUT"
+
+  fetch_tag:
+      needs: fetch_run_id
+      runs-on: ubuntu-latest
+
+      env:
+        RUN_ID: ${{ needs.fetch_run_id.outputs.run_id }}
+
+      outputs:
+        docker_image_tag: ${{ steps.get_docker_tag.outputs.docker_image_tag }}
+
+      steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Run ID from previous fetch_run_id job
+        run: echo Run ID from previous job ${{ env.RUN_ID }}
+        
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          # TODO: Create a token with basic repo permissions
+          name: docker-image-tag
+          github-token: ${{ secrets.GH_PAT_TAG }}
+          repository: e-mission/e-mission-server
+          run-id: ${{ env.RUN_ID }}
```

Comment:
As highlighted in this [comment in the documented issue](https://github.com/e-mission/e-mission-docs/issues/1048#issuecomment-2093776437), one of the final stages I had reached in designing the CI/CD was to eliminate having to manually update any docker tags (from a developer's perspective who will use the repos).

In the public-dash and admin-dash workflows, workflow dispatch directly receives it via input parameters, so it doesn't have to deal with artifacts.
But the push event is the one that utilizes the docker tags from the downloaded artifact. 
The tag from this artifact is the one that is written to the .env file for every push event.


------

Now a few scenarios:
1. What happens if a developer doesn't have .env file in their checkout out repo?
The .env file would always be present once this PR has been merged.
As a part of this PR, we are manually adding the .env file with the latest server image tag currently available in Dockerhub. This manual addition only needs to be done once during these code changes. Once implemented, the workflow will automatically update the .env file. 
Developers would then just pull in the remote changes and they should have the appropriate .env file.

2. What happens if a developer changes the image tag in .env file?
The workflow first executes the steps to fetch the latest tag from the server repo.
Any existing tag in .env file will be overridden with this latest tag.
This ensures two things: 
a. incorrect/non-existent tags will be overridden.
b. latest server image is used always.

------

To summarize, no one would need to update the .env file, the workflow would do it automatically.

Reply:
With that said, I do kind of see why we don't need to download the artifact.

Let's consider a few more scenarios:

1. Workflow dispatch: Server image is updated and new version pushed.
This will trigger the workflow dispatch which triggers workflows in the dashboard repos.
This updates the .env file.
The new tag is then used in building the image.

-----

2. Push event: Server image is not updated; dashboard repos have changes pushed.
In the existing workflow, the push will download the artifact and overwrite the .env file.

But, the .env file already has the latest tag from when the workflow dispatch event occurred. 
So, the push event can simply read in the latest server tag from the .env file.